stock_cycle_count/models/stock_cycle_count_rule.py

```diff
@@ -0,0 +1,269 @@
+            ("accuracy", _("Minimum Accuracy")),
+            ("zero", _("Zero Confirmation")),
+        ]
+
+    @api.constrains("rule_type", "warehouse_ids")
+    def _check_zero_rule(self):
+        for rec in self:
+            if rec.rule_type == "zero" and len(rec.warehouse_ids) > 1:
+                raise ValidationError(
+                    _(
+                        "Zero confirmation rules can only have one warehouse "
+                        "assigned."
+                    )
+                )
+            if rec.rule_type == "zero":
+                zero_rule = self.search(
+                    [
+                        ("rule_type", "=", "zero"),
+                        ("warehouse_ids", "=", rec.warehouse_ids.id),
+                    ]
+                )
+                if len(zero_rule) > 1:
+                    raise ValidationError(
+                        _(
+                            "You can only have one zero confirmation rule per "
+                            "warehouse."
+                        )
+                    )
+
+    @api.depends("rule_type")
+    def _compute_rule_description(self):
+        if self.rule_type == "periodic":
+            self.rule_description = _(
+                "Ensures that at least a defined number "
+                "of counts in a given period will "
+                "be run."
+            )
+        elif self.rule_type == "turnover":
+            self.rule_description = _(
+                "Schedules a count every time the total "
+                "turnover of a location exceeds the "
+                "threshold. This considers every "
+                "product going into/out of the location"
+            )
+        elif self.rule_type == "accuracy":
+            self.rule_description = _(
+                "Schedules a count every time the "
+                "accuracy of a location goes under a "
+                "given threshold."
+            )
+        elif self.rule_type == "zero":
+            self.rule_description = _(
+                "Perform an Inventory Adjustment every "
+                "time a location in the warehouse runs "
+                "out of stock in order to confirm it is "
+                "truly empty."
+            )
+        else:
+            self.rule_description = _("(No description provided.)")
+
+    @api.constrains("periodic_qty_per_period", "periodic_count_period")
+    def _check_negative_periodic(self):
+        for rec in self:
+            if rec.periodic_qty_per_period < 1:
+                raise ValidationError(
+                    _(
+                        "You cannot define a negative or null number of counts "
+                        "per period."
+                    )
+                )
+            if rec.periodic_count_period < 0:
+                raise ValidationError(_("You cannot define a negative period."))
+
+    @api.onchange("location_ids")
+    def _onchange_locaton_ids(self):
```

Comment:
Change warehouse_ids field to compute/store/readonly=False style and change this method to compute